packages/cli/test/integration/public-api/projects.test.ts

```diff
@@ -35,7 +35,7 @@ describe('Projects in Public API', () => {
 			/**
 			 * Act
 			 */
-			const response = await testServer.publicApiAgentFor(owner).get('/projects');
+			const response = await testServer.publicApiAgentWithApiKey(apiKey).get('/projects');
```

Comment:
We could have also kept the interface for this the same, i.e. `publicApiAgentFor(owner)`. In `createOwnerWithApiKey` create the owner and the API key. Return `User` so that it has the `apiKeys` field populated with the created API key. Then we could pass the `owner` to `publicApiAgentFor` method like before and in that method 1) use the API key from `owner.apiKeys` is it's there. If not, try to query it from DB. If there are none or multiple ones throw an error cause we don't know which one to use. We could also keep the `publicApiAgentWithApiKey` method so we can test the case where we provide an invalid API key.

IMHO the interface is a bit nicer that way. WDYT?

Reply:
Agreed it's a much better interface like this. Will update it.